src/thunderbots/software/test/ai/navigator/placeholder_navigator/placeholder_navigator.cpp

```diff
@@ -0,0 +1,194 @@
+}
+
+TEST(PlaceholderNavigatorTest, convert_move_intent_to_move_primitive)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+    PlaceholderNavigator placeholderNavigator;
+
+    std::vector<std::unique_ptr<Intent>> intents;
+    intents.emplace_back(
+        std::make_unique<MoveIntent>(0, Point(), Angle::quarter(), 0, 1));
+
+    auto primitive_ptrs = placeholderNavigator.getAssignedPrimitives(world, intents);
+
+    // Make sure we got exactly 1 primitive back
+    EXPECT_EQ(primitive_ptrs.size(), 1);
+
+    auto expected_primitive = MovePrimitive(0, Point(), Angle::quarter(), 0);
+    auto primitive          = dynamic_cast<MovePrimitive &>(*(primitive_ptrs.at(0)));
+    EXPECT_EQ(expected_primitive, primitive);
+}
+
+TEST(PlaceholderNavigatorTest, convert_movespin_intent_to_movespin_primitive)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+    PlaceholderNavigator placeholderNavigator;
+
+    std::vector<std::unique_ptr<Intent>> intents;
+    intents.emplace_back(
+        std::make_unique<MoveSpinIntent>(0, Point(), AngularVelocity::full(), 1));
+
+    auto primitive_ptrs = placeholderNavigator.getAssignedPrimitives(world, intents);
+
+    // Make sure we got exactly 1 primitive back
+    EXPECT_EQ(primitive_ptrs.size(), 1);
+
+    auto expected_primitive = MoveSpinPrimitive(0, Point(), AngularVelocity::full());
+    auto primitive          = dynamic_cast<MoveSpinPrimitive &>(*(primitive_ptrs.at(0)));
+    EXPECT_EQ(expected_primitive, primitive);
+}
+
+TEST(PlaceholderNavigatorTest, convert_pivot_intent_to_pivot_primitive)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+    PlaceholderNavigator placeholderNavigator;
+
+    std::vector<std::unique_ptr<Intent>> intents;
+    intents.emplace_back(
+        std::make_unique<PivotIntent>(0, Point(1, 0.4), Angle::half(), 3.2, 1));
+
+    auto primitive_ptrs = placeholderNavigator.getAssignedPrimitives(world, intents);
+
+    // Make sure we got exactly 1 primitive back
+    EXPECT_EQ(primitive_ptrs.size(), 1);
+
+    auto expected_primitive = PivotPrimitive(0, Point(1, 0.4), Angle::half(), 3.2);
+    auto primitive          = dynamic_cast<PivotPrimitive &>(*(primitive_ptrs.at(0)));
+    EXPECT_EQ(expected_primitive, primitive);
+}
+
+TEST(PlaceholderNavigatorTest, convert_stop_intent_to_stop_primitive)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+    PlaceholderNavigator placeholderNavigator;
+
+    std::vector<std::unique_ptr<Intent>> intents;
+    intents.emplace_back(std::make_unique<StopIntent>(0, false, 1));
+
+    auto primitive_ptrs = placeholderNavigator.getAssignedPrimitives(world, intents);
+
+    // Make sure we got exactly 1 primitive back
+    EXPECT_EQ(primitive_ptrs.size(), 1);
+
+    auto expected_primitive = StopPrimitive(0, false);
+    auto primitive          = dynamic_cast<StopPrimitive &>(*(primitive_ptrs.at(0)));
+    EXPECT_EQ(expected_primitive, primitive);
+}
```

Comment:
Could we add one test for multiple intents/primitives?

Reply:
test added